src/cpp/fastdds/domain/DomainParticipantImpl.hpp

```diff
@@ -504,6 +505,16 @@ class DomainParticipantImpl
      */
     bool has_active_entities();
 
+    ReturnCode_t enable_monitor_service() const;
+
+    ReturnCode_t disable_monitor_service() const;
+
+    ReturnCode_t check_compatible_qos(
+            const fastrtps::rtps::WriterProxyData& wdata,
+            const fastrtps::rtps::ReaderProxyData& rdata,
+            fastrtps::rtps::EDP::MatchingFailureMask& reason,
+            fastdds::dds::PolicyMask& incompatible_qos) const;
```

Comment:
Even though most of these methods are not documented and they are not part of the public API, the most recent criteria is to document every single method to help future developers.

Reply:
(I copy pasted the same as on the public api)